packages/did-comm/src/action-handler.ts

```diff
@@ -67,8 +130,355 @@ export class DIDComm implements IAgentPlugin {
     this.methods = {
-      sendMessageDIDCommAlpha1: this.sendMessageDIDCommAlpha1,
+      sendMessageDIDCommAlpha1: this.sendMessageDIDCommAlpha1.bind(this),
+      getDIDCommMessageMediaType: this.getDidCommMessageMediaType.bind(this),
+      unpackDIDCommMessage: this.unpackDIDCommMessage.bind(this),
+      packDIDCommMessage: this.packDIDCommMessage.bind(this),
+      sendDIDCommMessage: this.sendDIDCommMessage.bind(this),
+    }
+  }
+
+  /** {@inheritdoc IDIDComm.packDIDCommMessage} */
+  async packDIDCommMessage(
+    args: IPackDIDCommMessageArgs,
+    context: IAgentContext<IDIDManager & IKeyManager & IResolver>,
+  ): Promise<IPackedDIDCommMessage> {
+    switch (args.packing) {
+      case 'authcrypt': // intentionally omitting break
+      case 'anoncrypt':
+        return this.packDIDCommMessageJWE(args, context)
+      case 'none':
+        const message = {
+          ...args.message,
+          typ: DIDCommMessageMediaType.PLAIN,
+        }
+        return { message: JSON.stringify(message) }
+      case 'jws':
+        return this.packDIDCommMessageJWS(args, context)
+      default:
+        throw new Error(`not_implemented: packing messages as ${args.packing} is not supported yet`)
+    }
+  }
+
+  private async packDIDCommMessageJWS(
+    args: IPackDIDCommMessageArgs,
+    context: IAgentContext<IDIDManager & IKeyManager & IResolver>,
+  ): Promise<IPackedDIDCommMessage> {
+    const message = args.message
+    let keyRef: string | undefined = args.keyRef
+    let kid: string
+    // check that the message has from field that is managed
+    let managedSender: IIdentifier | undefined
+    try {
+      managedSender = await context.agent.didManagerGet({ did: message.from || '' })
+    } catch (e) {
+      debug(`message.from(${message.from}) is not managed by this agent`)
+    }
+    if (!message.from || !isDefined(managedSender)) {
+      throw new Error('invalid_argument: `from` field must be a DID managed by this agent')
+    }
+
+    // obtain sender signing key(s) from authentication section
+    const senderKeys = await mapIdentifierKeysToDoc(managedSender, 'authentication', context)
+    // try to find a managed signing key that matches keyRef
+    let signingKey = null
+    if (isDefined(keyRef)) {
+      signingKey = senderKeys.find((key) => key.kid === keyRef || key.meta.verificationMethod.id === keyRef)
+    }
+    // otherwise use the first available one.
+    signingKey = signingKey ? signingKey : senderKeys[0]
+
+    if (!signingKey) {
+      throw new Error(`key_not_found: could not locate a suitable signing key for ${message.from}`)
+    } else {
+      kid = signingKey.meta.verificationMethod.id
+    }
+    let alg: string
+    if (signingKey.type === 'Ed25519') {
+      alg = 'EdDSA'
+    } else if (signingKey.type === 'Secp256k1') {
+      alg = 'ES256K'
+    } else {
+      throw new Error(
+        `not_supported: key of type ${signingKey.type} is not supported for JWS didcomm message`,
+      )
+    }
```

Comment:
is this needed because the didcomm spec only allows a certain set of signature algs?

Reply:
I think it's more a limitation of Veramo supported keys and algorithms.
For the moment there is a 1:1 correspondence between the key types and the JWA. If this changes, we would need to require an `alg` parameter to allow the user to specify